newIDE/app/src/Profile/Subscription/CancelReasonDialog.js

```diff
@@ -0,0 +1,270 @@
+// @flow
+import { Trans, t } from '@lingui/macro';
+import { I18n } from '@lingui/react';
+import * as React from 'react';
+import Dialog, { DialogPrimaryButton } from '../../UI/Dialog';
+import AuthenticatedUserContext from '../AuthenticatedUserContext';
+import { changeUserSubscription } from '../../Utils/GDevelopServices/Usage';
+import { ColumnStackLayout, LineStackLayout } from '../../UI/Layout';
+import useAlertDialog from '../../UI/Alert/useAlertDialog';
+import GDevelopGLogo from '../../UI/CustomSvgIcons/GDevelopGLogo';
+import Form from '../../UI/Form';
+import Text from '../../UI/Text';
+import { Spacer } from '../../UI/Grid';
+import TextField from '../../UI/TextField';
+import Checkbox from '../../UI/Checkbox';
+import FlatButton from '../../UI/FlatButton';
+import StarIcon from '../../UI/CustomSvgIcons/Star';
+
+type Props = {|
+  onClose: Function,
+  onCloseAfterSuccess: Function,
+|};
+
+const CancelReasonDialog = ({ onClose, onCloseAfterSuccess }: Props) => {
+  const [isCancelingSubscription, setIsCancelingSubscription] = React.useState(
+    false
+  );
+  const [hasCanceledSubscription, setHasCanceledSubscription] = React.useState(
+    false
+  );
+  const [
+    stoppedMakingGamesChecked,
+    setStoppedMakingGamesChecked,
+  ] = React.useState(false);
+  const [strugglingChecked, setStrugglingChecked] = React.useState(false);
+  const [
+    preferFreeVersionChecked,
+    setPreferFreeVersionChecked,
+  ] = React.useState(false);
+  const [missingFeatureChecked, setMissingFeatureChecked] = React.useState(
+    false
+  );
+  const [qualityIssuesChecked, setQualityIssuesChecked] = React.useState(false);
+  const [otherChecked, setOtherChecked] = React.useState(false);
+  const [freeText, setFreeText] = React.useState('');
+  const authenticatedUser = React.useContext(AuthenticatedUserContext);
+  const { showAlert } = useAlertDialog();
+
+  const canSubmit = freeText.trim().length > 0;
```

Comment:
The free text is always required? Could/should we allow not to enter anything in the case of a checkbox other than "other reason" being selected?

Reply:
I thought making it always required, but will change it then (except "missing feature" and "other reason")